docs/contributing/documentation_guidelines.md

```diff
@@ -0,0 +1,298 @@
+
+    ```python
+    INSTALLED_APPS = [
+        ...
+        "wagtail.core",
+        ...
+    ]
+    ```
+
+### Links
+
+Links are fundamental in documentation. Use internal links to tie your content to other docs, and external links as needed. Pick relevant text for links, so readers know where they will land.
+
+Don’t rely on [`links over code`](https://www.example.com/), as they are impossible to spot.
+
+```md
+An [external link](https://wwww.example.com).
+An [internal link to another document](/reference/contrib/legacy_richtext.md).
+A [link to a reference](register_reports_menu_item).
+```
+
+<details>
+
+<summary>Rendered output</summary>
+
+An [external link](https://wwww.example.com).
+An [internal link to another document](/reference/contrib/legacy_richtext.md).
+A [link to a reference](register_reports_menu_item).
+
+</details>
+
+Reference links rely on creating a reference in reStructuredText. Prefer linking to the whole document if at all possible, otherwise create a reference by embedding reStructuredText with `eval_rst`:
+
+    ```eval_rst
+    .. _register_reports_menu_item:
+    ```
+
+### Note and warning call-outs
+
+Use notes and warnings sparringly, as they rely on reStructuredText syntax that is harder to author.
+
+    ```eval_rst note:: Notes can provide complementary information.
+    ```
+
+    ```eval_rst warning:: Warnings can be scary.
+    ```
+
+<details>
+
+<summary>Rendered output</summary>
+
+```eval_rst note:: Notes can provide complementary information.
+```
+
+```eval_rst warning:: Warnings can be scary.
+```
+
+</details>
+
+### Images
+
+Images are hard to keep up-to-date as documentation evolves, but can be worthwhile nonetheless. Here are guidelines when adding images:
+
+- All images should have meaningful [alt text](https://axesslab.com/alt-texts/) unless they are decorative.
+- Images are served as-is – pick the correct format, and losslessly compress all images.
+- Use absolute paths for image files so they are more portable.
+
+```md
+![Screenshot of the workflow editing interface, with fields to change the workflow name, tasks, and assigned pages](/_static/images/screen44_workflow_edit.png)
+```
+
+<details>
+
+<summary>Rendered output</summary>
+
+![Screenshot of the workflow editing interface, with fields to change the workflow name, tasks, and assigned pages](/_static/images/screen44_workflow_edit.png)
+
+</details>
+
+### Autodoc
+
+With its [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) feature, Sphinx supports writing documentation in Python docstrings for subsequent integration in the project’s documentation pages. This is a very powerful feature which we highly recommend using to document Wagtail’s APIs.
+
+    ```eval_rst
+    .. module:: wagtail.tests.testapp.views
```

Comment:
It might be nice to use a function that has a docstring ([arbitrary example](https://github.com/wagtail/wagtail/blob/bf33842f450ef9270a3cf258dafc2b8303b4913b/wagtail/core/utils.py#L91)), to further demonstrate how this can pull them in.

Reply:
Indeed!